### PR TITLE
add components diagram

### PR DESCRIPTION
**Pull request by skorupskimat**

### CODE REVIEW

File: violetplugin-componentdiagram/src/main/java/com/horstmann/violet/product/diagram/component/node/ComponentNode.java [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+    @Override
+    public void setTextColor(Color textColor)
+    {
+        text.setTextColor(textColor);
+        super.setTextColor(textColor);
+    }
+
+    /**
+     *
+     * @return Tooltip string.
```
Comment (nifrez) [inline violetplugin-componentdiagram/src/main/java/com/horstmann/violet/product/diagram/component/node/ComponentNode.java]: Useless javadoc

File: violetplugin-componentdiagram/src/main/java/com/horstmann/violet/product/diagram/component/node/ComponentNode.java [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    }
+
+    @Override
+    public LineText getMethods()
+    {
+        return null;
+    }
+
+    /**
+     * Gets connection point.
```
Comment (nifrez) [inline violetplugin-componentdiagram/src/main/java/com/horstmann/violet/product/diagram/component/node/ComponentNode.java]: Useless javadoc / not completed javadoc

File: violetplugin-componentdiagram/src/main/java/com/horstmann/violet/product/diagram/component/node/ComponentNode.java [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+                    connectionPoint.getY()
+            );
+        }
+
+        return connectionPoint;
+    }
+
+    /**
+     * Adds connection edge.
+     * @param edge
```
Comment (nifrez) [inline violetplugin-componentdiagram/src/main/java/com/horstmann/violet/product/diagram/component/node/ComponentNode.java]: useless javadoc

File: violetplugin-componentdiagram/src/main/java/com/horstmann/violet/product/diagram/component/node/ComponentNode.java [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+     *
+     * @param newValue the new text description
+     */
+    public void setText(LineText newValue)
+    {
+        text.setText(newValue.toEdit());
+    }
+
+    /**
+     * Gets the text property value.
```
Comment (nifrez) [inline violetplugin-componentdiagram/src/main/java/com/horstmann/violet/product/diagram/component/node/ComponentNode.java]: useless javadoc

File: violetplugin-componentdiagram/test/com/horstmann/violet/product/diagram/component/node/ComponentNodeTest.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+
+    @Before
+    public void setUp() throws Exception {
+         componentNode = new ComponentNode();
+         componentNode.setText(new SingleLineText(LineText.DEFAULT_CONVERTER));
+    }
+    @Test
+    public void beforeReconstruction() throws Exception {
+
```
Comment (nifrez) [inline violetplugin-componentdiagram/test/com/horstmann/violet/product/diagram/component/node/ComponentNodeTest.java]: ??????????????????

File: violetplugin-componentdiagram/test/com/horstmann/violet/product/diagram/component/node/ComponentNodeTest.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @Test
+    public void copy() throws Exception {
+
```
Comment (nifrez) [inline violetplugin-componentdiagram/test/com/horstmann/violet/product/diagram/component/node/ComponentNodeTest.java]: ?????????????????

File: violetplugin-componentdiagram/test/com/horstmann/violet/product/diagram/component/node/ComponentNodeTest.java [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @Test
+    public void createContentStructure() throws Exception {
+
```
Comment (nifrez) [inline violetplugin-componentdiagram/test/com/horstmann/violet/product/diagram/component/node/ComponentNodeTest.java]: ?????????????????

File: violetplugin-componentdiagram/test/com/horstmann/violet/product/diagram/component/node/ComponentNodeTest.java [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @Test
+    public void setTextColor() throws Exception {
+
```
Comment (nifrez) [inline violetplugin-componentdiagram/test/com/horstmann/violet/product/diagram/component/node/ComponentNodeTest.java]: ?????????????????

File: violetplugin-componentdiagram/test/com/horstmann/violet/product/diagram/component/node/ComponentNodeTest.java [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @Test
+    public void getToolTip() throws Exception {
+
```
Comment (nifrez) [inline violetplugin-componentdiagram/test/com/horstmann/violet/product/diagram/component/node/ComponentNodeTest.java]: ?????????????????

File: violetplugin-componentdiagram/test/com/horstmann/violet/product/diagram/component/node/ComponentNodeTest.java [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @Test
+    public void getConnectionPoint() throws Exception {
+
```
Comment (nifrez) [inline violetplugin-componentdiagram/test/com/horstmann/violet/product/diagram/component/node/ComponentNodeTest.java]: ?????????????????

File: violetplugin-componentdiagram/test/com/horstmann/violet/product/diagram/component/node/ComponentNodeTest.java [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @Test
+    public void addConnection() throws Exception {
+
```
Comment (nifrez) [inline violetplugin-componentdiagram/test/com/horstmann/violet/product/diagram/component/node/ComponentNodeTest.java]: ?????????????????

File: violetplugin-componentdiagram/test/com/horstmann/violet/product/diagram/component/node/ComponentNodeTest.java [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @Test
+    public void setText() throws Exception {
+
```
Comment (nifrez) [inline violetplugin-componentdiagram/test/com/horstmann/violet/product/diagram/component/node/ComponentNodeTest.java]: ?????????????????

File: violetplugin-componentdiagram/test/com/horstmann/violet/product/diagram/component/node/ComponentNodeTest.java [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @Test
+    public void getText() throws Exception {
+
```
Comment (nifrez) [inline violetplugin-componentdiagram/test/com/horstmann/violet/product/diagram/component/node/ComponentNodeTest.java]: ?????????????????

**Review by nifrez — COMMENTED** [review]
Remove empty tests, and useless javadocs

File: violetplugin-componentdiagram/src/main/java/com/horstmann/violet/product/diagram/component/ComponentDiagramGraph.java [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+import com.horstmann.violet.product.diagram.component.node.ComponentNode;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * A UML component diagram.
+ */
+public class ComponentDiagramGraph extends AbstractGraph {
```
Comment (abizukojc) [inline violetplugin-componentdiagram/src/main/java/com/horstmann/violet/product/diagram/component/ComponentDiagramGraph.java]: Bad formatting. The bracket { should be under the class/method header. Correct it everywhere in this PR.

File: pom.xml [hunk 15]
```diff
@@ -3,87 +3,93 @@
...
+						<source>1.8</source>
+						<target>1.8</target>
+						<encoding>ISO-8859-1</encoding>
+					</configuration>
+				</plugin>
+			</plugins>
+		</pluginManagement>
+	</build>
 
     <!-- Modules 'exe' and 'web' are commented out because they're causing problems during project build-->
```
Comment (abizukojc) [inline pom.xml]: This block of comments is useless now.

File: violetplugin-componentdiagram/test/com/horstmann/violet/product/diagram/component/node/ComponentNodeTest.java (hunk 13 above)
Comment (bartoszsledz) [reply violetplugin-componentdiagram/test/com/horstmann/violet/product/diagram/component/node/ComponentNodeTest.java]: xDDDDDDDDDDDDDDDDDDDDDDDDDDD no przeciez przechodzą 🥇

Comment (skorupskimat) [reply violetplugin-componentdiagram/test/com/horstmann/violet/product/diagram/component/node/ComponentNodeTest.java]: no jak to nie ma testów?

File: violetplugin-componentdiagram/src/main/java/com/horstmann/violet/product/diagram/component/node/ComponentNode.java [hunk 16]
```diff
@@ -0,0 +1,256 @@
...
+
+    /**
+     * Adds connection edge.
+     */
+    @Override
+    public boolean addConnection(IEdge edge)
+    {
+        if (edge.getEndNode() != null && this != edge.getEndNode())
+        {
+            return true;
```
Comment (NdV66) [inline violetplugin-componentdiagram/src/main/java/com/horstmann/violet/product/diagram/component/node/ComponentNode.java]: I don't like double returns in one method.
It should be better, if you change it to one return.
boolean result = false;
if(...) {
result = true;
}
return result;

**Comment by NdV66** [comment]
@RoughTomato @abizukojc 

It **ISN'T** able to be merge yet 👎 
We'll waiting for changes.

**Review by abizukojc — COMMENTED** [review]
Re-check the comments

**Review by orchowski — APPROVED** [review]

**Review by abizukojc — CHANGES_REQUESTED** [review]
Check the comments once again.

**Comment by NdV66** [comment]
@abizukojc please review

**Review by abizukojc — CHANGES_REQUESTED** [review]
There are still comments in this PR. Fix commented places, or reply the comments,

**Comment by RoughTomato** [comment]
outdated files ../../../../ComponentNodeTest.java might cause compatibility problems. Please update to current HEAD~.

**Comment by NdV66** [comment]
Yes, please update.

**Review by abizukojc — APPROVED** [review]

**Pull request merged**